Update nullability info (#563)
vaadin/docs: articles/fusion/advanced/endpoints-generator.asciidoc

NOTE: For more information about type mapping between Java and TypeScript, please refer to <<type-conversion#, type conversion page>>.

=== Type Nullability

> Types can be either nullable (optional) or non-nullable (required).
> By default, types are mapped and generated using the Java rules:
> 
> - Any primitive type (e.g., `int`) is non-nullable.
> - Any reference type (e.g., `String` or `Integer`) is nullable.
> - A collection accepts `null` unless the collection item type is primitive.
> - A map accepts `null` unless the collection item type is primitive.
> 
> To make any type non-nullable, you need to put a `@Nonnull` annotation on it.

Reviewer: ```suggestion
Types are either _nullable_ (optional) or _non-nullable_ (required).
By default, types are mapped and generated using the Java rules:

- Any primitive type, such as `int`, is non-nullable.
- Any reference type, such as `String` or `Integer`, is nullable.
- A collection accepts `null` unless the collection item type is primitive.
- A map accepts `null` unless the collection item type is primitive.

To make a type non-nullable, you need to put a `@Nonnull` annotation on it.
```
